doc/features/open-telemetry/tracing-rfc.md

```diff
@@ -0,0 +1,414 @@
+
+According to the specification, the span name "SHOULD be set to a low cardinality value representing the statement executed on the database. It MAY be a stored procedure name (without arguments), DB statement without variable arguments, operation name, etc.".\
+The specification also (and only) specifies the span name for SQL databases:\
+"Since SQL statements may have very high cardinality even without arguments, SQL spans SHOULD be named the following way, unless the statement is known to be of low cardinality:\
+`<db.operation> <db.name>.<db.sql.table>`, provided that `db.operation` and `db.sql.table` are available. If `db.sql.table` is not available due to its semantics, the span SHOULD be named `<db.operation> <db.name>`. It is not recommended to attempt any client-side parsing of `db.statement` just to get these properties,
+they should only be used if the library being instrumented already provides them. When it's otherwise impossible to get any meaningful span name, `db.name` or the tech-specific database name MAY be used."
+
+To avoid parsing the statement, the **span name** in this implementation has the **keyspace name** as its value.
+
+### Span attributes
+
+This implementation includes, by default, the **required** attributes for Database, and Cassandra spans.\
+`server.address` and `server.port`, despite only **recommended**, are included to give information regarding the client connection.\
+`db.statement` is optional given that this attribute may contain sensitive information:
+
+| Attribute  | Description  | Type | Required | Supported Values|
+|---|---|---|---|---|
+| span.kind | Describes the relationship between the Span, its parents, and its children in a Trace. | string | true | client |
+| db.system | An identifier for the database management system (DBMS) product being used. | string | true | cassandra |
+| db.name | The keyspace name in Cassandra. | string | true | *keyspace in use* |
+| db.statement | The database statement being executed. | string | false | *database statement in use* |
+| server.address | Name of the database host. | string | true | e.g.: example.com; 10.1.2.80; /tmp/my.sock |
+| server.port | Server port number. Used in case the port being used is not the default | int | false | e.g.: 9445 |
+
+## Usage
+
+### Package installation
+
+The OpenTelemetry implementation is included in the package `CassandraCSharpDriver.OpenTelemetry`.
+
+### Exporting Cassandra activity
+
+The extension method `AddOpenTelemetryInstrumentation()` is available in the cluster builder, so the activity can be exported for database operations:
+
+```csharp
+var cluster = Cluster.Builder()
+    .AddContactPoint("127.0.0.1")
+    .WithSessionName("session-name")
+    .AddOpenTelemetryInstrumentation()
```

Comment:
The open-telemetry-contrib for [cassandra](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Cassandra) already defines a `.WithOpenTelemetryMetrics()`.  By naming this `.AddOpenTelemetryInstrumentation()` couldn't this mistakenly imply that it will enable all the instrumentation for Cassandra (metrics, tracing)?

Reply:
Good question. Metrics are not being exported or created by any standard and, in [future possibilities](https://github.com/simaoribeiro/csharp-driver/blob/793dfec7b6262208094b14c133b42871060e3780/doc/features/open-telemetry/tracing-rfc.md#future-possibilities) I mentioned that metrics could be exported from OTEL in the future.
However, it makes sense that the extension method should have a more explicit name. Maybe `AddOpenTelemetryTraceInstrumentation()`? What do you say?

Reply:
Done, please review

Reply:
I agree with `AddOpenTelemetryTraceInstrumentation` in order to prevent miss judgement around what that is doing. Unless the implementation of other instrumentation features are planned for the "near" future.

Nevertheless, their input on this is going to be more valuable to understand the best naming.